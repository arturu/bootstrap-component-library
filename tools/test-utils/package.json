{
  "name": "@openeuropa/bcl-test-utils",
  "version": "0.1.3",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
