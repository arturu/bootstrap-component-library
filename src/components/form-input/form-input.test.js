import { merge, renderTwigFileAsNode } from "@oe/test-utils";

import demoData from "@oe/data-form-input/data";

const template = "@oe/form-input/form-input.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Form Input", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, placeholder: "Text here" })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly when required", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, required: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly when disabled", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, disabled: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly when readonly", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, readonly: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly when invalid", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, invalid: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with floating label", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, floating: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly when horizontal", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, horizontal: true })
    ).resolves.toMatchSnapshot();
  });

  test("checkbox renders correctly", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, type: "checkbox" })
    ).resolves.toMatchSnapshot();
  });

  test("radio renders correctly", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, type: "radio" })
    ).resolves.toMatchSnapshot();
  });

  test("switch renders correctly", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, type: "checkbox", switch: true })
    ).resolves.toMatchSnapshot();
  });
});
