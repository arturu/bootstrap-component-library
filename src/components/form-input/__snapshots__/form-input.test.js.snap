// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Form Input checkbox renders correctly 1`] = `
<div
  class="form-check"
>
  <input
    aria-describedby="helperText"
    class="form-check-input"
    id="example-form-input-1"
    type="checkbox"
  />
  <label
    class="form-check-label"
    for="example-form-input-1"
  >
    Example form input
  </label>
  <div
    class="form-text"
    id="helperText"
  >
    Helper text for the form input
  </div>
</div>
`;

exports[`OE - Form Input radio renders correctly 1`] = `
<div
  class="form-check"
>
  <input
    aria-describedby="helperText"
    class="form-check-input"
    id="example-form-input-1"
    type="radio"
  />
  <label
    class="form-check-label"
    for="example-form-input-1"
  >
    Example form input
  </label>
  <div
    class="form-text"
    id="helperText"
  >
    Helper text for the form input
  </div>
</div>
`;

exports[`OE - Form Input renders correctly 1`] = `
<label
  class="form-label"
  for="example-form-input-1"
>
  Example form input
</label>
`;

exports[`OE - Form Input renders correctly when disabled 1`] = `
<label
  class="form-label"
  for="example-form-input-1"
>
  Example form input
</label>
`;

exports[`OE - Form Input renders correctly when horizontal 1`] = `
<div
  class="row"
>
  <label
    class="col-form-label col-sm-2"
    for="example-form-input-1"
  >
    Example form input
  </label>
  <div
    class="col-sm-10"
  >
    <input
      aria-describedby="helperText"
      class="form-control form-control-sm"
      id="example-form-input-1"
      type="text"
    />
  </div>
  <div
    class="form-text"
    id="helperText"
  >
    Helper text for the form input
  </div>
</div>
`;

exports[`OE - Form Input renders correctly when invalid 1`] = `
<label
  class="form-label"
  for="example-form-input-1"
>
  Example form input
</label>
`;

exports[`OE - Form Input renders correctly when readonly 1`] = `
<label
  class="form-label"
  for="example-form-input-1"
>
  Example form input
</label>
`;

exports[`OE - Form Input renders correctly when required 1`] = `
<label
  class="form-label"
  for="example-form-input-1"
>
  Example form input
</label>
`;

exports[`OE - Form Input renders correctly with floating label 1`] = `
<div
  class="form-floating"
>
  <input
    aria-describedby="helperText"
    class="form-control form-control-sm"
    id="example-form-input-1"
    type="text"
  />
  <label
    class="form-label"
    for="example-form-input-1"
  >
    Example form input
  </label>
  <div
    class="form-text"
    id="helperText"
  >
    Helper text for the form input
  </div>
</div>
`;

exports[`OE - Form Input switch renders correctly 1`] = `
<div
  class="form-check form-switch"
>
  <input
    aria-describedby="helperText"
    class="form-check-input"
    id="example-form-input-1"
    type="checkbox"
  />
  <label
    class="form-check-label"
    for="example-form-input-1"
  >
    Example form input
  </label>
  <div
    class="form-text"
    id="helperText"
  >
    Helper text for the form input
  </div>
</div>
`;
